### DISCUSSION TITLE
When manually creating Document with value toString() returns null

### DISCUSSION THREAD

**Discussion by dovidgef** [eemeli/yaml#217] [Q&A] [answered]
There appears to be a bug when creating a `YAML.Document` object with an initial value.
The documentation seems to indicate that the Document constructer takes three possible arguments:
`new YAML.Document(value, replacer?, options = {})`

But in reality it appears that it only takes one argument `options`.
And when I set a value as the first argument calling `toString()` returns `null`.

e.g.
```
let yaml_doc = new YAML.Document(js_object, options);
console.log(yaml_doc.toString()) // outputs null
```

**Comment by dovidgef** [comment]
In the meantime in order to work around this issue, I needed to first use `YAML.stringify()` method on my object followed by `new YAML.parseDocument()` like so:
```
let yaml_doc = new YAML.parseDocument(YAML.stringify(obj));
```

**Comment by eemeli** [comment]
Could you check which version of `yaml` you're using? The v2 API is different from v1.

**Comment by dovidgef** [comment]
It appears that I am using `"yaml": "^1.10.0"`
This is the package version that was installed by default using NPM, the
documentation is for version 2?

**Comment by dovidgef** [comment]
I see now that I was looking at version 2 of the documentation.
While version 1 is still the default, shouldnt the documentation default to version 1 as well?

**Comment by dovidgef** [comment]
What is the correct way to create a YAML Document object from a JS object in version 1?

**Comment by eemeli** [comment, marked as answer]
This section of the v1 docs provides a couple of examples: https://eemeli.org/yaml/v1/#creating-nodes
